services/brig/src/Brig/API/MLS/KeyPackages.hs

```diff
@@ -124,17 +127,20 @@ claimLocalKeyPackages qusr skipOwn suite target = do
     assertUserNotLegalholded :: ExceptT ClientError (AppT r) ()
     assertUserNotLegalholded = do
       -- this is okay because there can only be one StoredUser per UserId
-      [su] <- lift $ liftSem $ getUsers [tUnqualified target]
-      for_ su.teamId \tid -> do
-        resp <- lift $ liftSem $ getUserLegalholdStatus target tid
-        -- if an admin tries to put a user under legalhold
-        -- the user has to first reject to be put under legalhold
-        -- before they can join conversations again
-        case resp.ulhsrStatus of
-          UserLegalHoldPending -> throwE ClientLegalHoldIncompatible
-          UserLegalHoldEnabled -> throwE ClientLegalHoldIncompatible
-          UserLegalHoldDisabled -> pure ()
-          UserLegalHoldNoConsent -> pure ()
+      mSu <- lift $ liftSem $ getUser (tUnqualified target)
+      case mSu of
+        Nothing -> pure () -- No user, no conflict.
```

Comment:
when is it a legal state to not have a stored user for a user id here?

Reply:
we might race an account deletion here. not likely, but doesn't do much damage to handle it, no?

Reply:
Then it’s better to get a 500 here than some undefined behaviour from further down?